core/base-service/route.js

```diff
@@ -15,9 +15,11 @@ const routeSchema = Joi.object({
     is: Joi.string().required(),
     then: Joi.array().items(Joi.string().required()),
   }),
+  queryParamSchema: Joi.object().schema(),
   queryParams: Joi.array().items(Joi.string().required()),
 })
   .xor('pattern', 'format')
+  .oxor('queryParamSchema', 'queryParams')
```

Comment:
Is this just because some services are still specifying query params via the `queryParams` array, or will the `queryParams` continue to live on long term?

Reply:
Originally I'd thought this was needed for legacy services, though I now realize that's not true. So it could be removed once all the services are all cleaned up.

Reply:
👍